java/core/src/main/java/co/worklytics/psoxy/gateway/impl/CommonRequestHandler.java

```diff
@@ -0,0 +1,209 @@
+package co.worklytics.psoxy.gateway.impl;
+
+import co.worklytics.psoxy.*;
+import co.worklytics.psoxy.gateway.*;
+import co.worklytics.psoxy.impl.SanitizerImpl;
+import co.worklytics.psoxy.rules.RulesUtils;
+import co.worklytics.psoxy.utils.URLUtils;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.auth.Credentials;
+import com.google.auth.http.HttpCredentialsAdapter;
+import com.google.common.net.MediaType;
+import lombok.*;
+import lombok.extern.java.Log;
+import org.apache.commons.lang3.tuple.Pair;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.net.URL;
+import java.util.*;
+import java.util.logging.Level;
+import java.util.stream.Collectors;
+
+@NoArgsConstructor(onConstructor_ = @Inject)
+@Log
+public class CommonRequestHandler {
+
+    //we have ~540 total in Cloud Function connection, so can have generous values here
+    final int SOURCE_API_REQUEST_CONNECT_TIMEOUT = 30_000;
+    final int SOURCE_API_REQUEST_READ_TIMEOUT = 300_000;
+
+    @Inject ConfigService config;
+    @Inject RulesUtils rulesUtils;
+    @Inject SourceAuthStrategy sourceAuthStrategy;
+    @Inject ObjectMapper objectMapper;
+    @Inject SanitizerFactory sanitizerFactory;
+    @Inject Rules rules;
+
+    private Sanitizer sanitizer;
```

Comment:
Variable is global for the class, but only used in `handle` L48, L113 and in L114. Maybe it can be local for the method

Reply:
yeah - tbd; we ideally want the instance of `CommonRequestHandler` to be re-used rather than re-instantiated for each request - and `Sanitizer` as well. We need to performance test